realsense/plugin.cpp

```diff
@@ -109,36 +218,66 @@ class realsense : public plugin {
-                    std::optional<cv::Mat *> img0 = std::nullopt;
-                    std::optional<cv::Mat *> img1 = std::nullopt;
-                    std::optional<cv::Mat *> rgb = std::nullopt;
-                    std::optional<cv::Mat *> depth = std::nullopt;
-                    if (last_iteration_cam != cam_type_.iteration)
-                    {
-                        last_iteration_cam = cam_type_.iteration;
-                        img0 = cam_type_.img0;
-                        img1 = cam_type_.img1;
-                        rgb = cam_type_.rgb;
-                        depth = cam_type_.depth;
+                     // Images
+                    if (cam_select == D4XXI){
+                        std::optional<cv::Mat> img0 = std::nullopt;
+                        std::optional<cv::Mat> img1 = std::nullopt;
+                        std::optional<cv::Mat> rgb = std::nullopt;
+                        std::optional<cv::Mat> depth = std::nullopt;
+                        if (last_iteration_cam != cam_D4XX_.iteration)
+                        {
+                            last_iteration_cam = cam_D4XX_.iteration;
+                            img0 = cam_D4XX_.img0;
+                            img1 = cam_D4XX_.img1;
+                            rgb = cam_D4XX_.rgb;
+                            depth = cam_D4XX_.depth;
+                        }
+                        
+                        // Submit to switchboard
+                        _m_imu_cam.put(_m_imu_cam.allocate<imu_cam_type>(
+                            imu_cam_type{
+                                imu_time_point,
+                                av,
+                                la,
+                                img0,
+                                img1,
+                                imu_time
+                            }
+                        ));
+                        
+                        if (rgb && depth)
+                        {
+                            _m_rgb_depth.put(_m_rgb_depth.allocate<rgb_depth_type>(
+                                rgb_depth_type{
+                                    rgb,
+                                    depth,
+                                    imu_time
+                                }
+                            ));
+                        }
                     }
 
-                    // Submit to switchboard
-                    _m_imu_cam.put(_m_imu_cam.allocate<imu_cam_type>(
-                        imu_time_point,
-                        av,
-                        la,
-                        img0,
-                        img1,
-                        imu_time,
-                    ));
-
-                    if (rgb && depth)
-                    {
-                        _m_rgb_depth.put(_m_rgb_depth.allocate<rgb_depth_type>(
-                            rgb,
-                            depth,
-                            imu_time,
+                    else if (cam_select == T26X){
```

Comment:
I think the code can be simplified a bit by declaring all four `std::optional<cv::Mat>` regardless of underlying camera type, but we can check `rgb && depth` and know if there is nothing there. This removes this conditional block and the need for `cam_type_${hardware}`. The earlier in code we "unify" the behavior for underlying hardware the better.